pretyping/structures.ml

```diff
@@ -155,6 +155,14 @@ let equal env p1 p2 = match p1, p2 with
   | Default_cs, Default_cs -> true
   | _ -> false
```

Comment:
`ValuePattern.equal` above is not used anymore. We can probably remove it.

BTW, I don't get why it requires `env` as an argument. If there is a good reason to do so, this PR may not be the right change to do.

Reply:
It is due to the fact that we needed to check equality of global references up to canonical kernel names.

Reply:
But `Environ.QGlobRef.equal` and `Environ.QProjection.Repr.equal` actually discard the first argument `env`. Do you mean there is a plan to change them not to discard `env`?

Reply:
Yes, the goal is ultimately to get rid of the dual name mechanism and replace it by an explicit quotient in the environment.

Reply:
Then entries of `GlobRef.Map` and `PatMap` (introduced in this PR) should be indexed by canonical names, right? If so, I'm totally fine with closing this PR for now and reworking it after that large refactoring work, since it does not improve anything in benchmarks.